x-pack/plugins/ml/public/application/datavisualizer/stats_table/types/field_vis_config.ts

```diff
@@ -70,3 +70,22 @@ export interface FieldVisConfig {
   fieldFormat?: any;
   isUnsupportedType?: boolean;
 }
+
+export interface FileBasedFieldVisConfig {
+  type: MlJobFieldType;
+  fieldName?: string;
+  stats?: FieldVisStats;
+  format?: string;
+}
+
+export function isFileBasedFieldVisConfig(
+  field: FieldVisConfig | FileBasedFieldVisConfig
+): field is FileBasedFieldVisConfig {
+  return !field.hasOwnProperty('existsInDocs');
+}
+
+export function isIndexBasedFieldVisConfig(
+  field: FieldVisConfig | FileBasedFieldVisConfig
+): field is FieldVisConfig {
+  return (field as FieldVisConfig).existsInDocs !== undefined;
```

Comment:
not sure if manual typecasting is a good approach in type guards

Reply:
Updated here https://github.com/elastic/kibana/pull/87598/commits/e6f1c5f39d01540e0a24875062722a36360390f1